packages/@expo/cli/src/start/server/metro/MetroBundlerDevServer.ts

```diff
@@ -15,16 +15,19 @@ import {
   sourceMapGeneratorNonBlocking,
   type SourceMapGeneratorOptions,
 } from 'metro/src/DeltaBundler/Serializers/sourceMapGenerator';
+import type MetroHmrServer from 'metro/src/HmrServer';
+import { type Client as MetroHmrClient } from 'metro/src/HmrServer';
```

Comment:
If there are only types imported, I would recommend using this notation. This also makes possible dependency check failures a warning only, as it's only importing types (not mixed imports).

```suggestion
import type { Client as MetroHmrClient } from 'metro/src/HmrServer';
```